packages/yarnpkg-fslib/sources/index.ts

```diff
@@ -396,4 +392,26 @@ export const xfs: XFS = Object.assign(new NodeFS(), {
       }
     }
   },
+
+  async rmtempPromise() {
+    await Promise.all(Array.from(tmpdirs.values()).map(async p => {
+      tmpdirs.delete(p);
```

Comment:
I'd move that after the `xfs.removePromise`. This way, even if it doesn't get removed right now, it *might* get removed once the program exits.

Reply:
I kept it like that since the original sync implementation in `process.on('exit')` did it this way, so i'm fine with either way.